### PR TITLE
Random circuits bristlecone

### PR DESCRIPTION
**Pull request by sboixo**
Generates random circuits (supremacy v2) on Bristlecone

### CODE REVIEW

File: cirq/experiments/google_v2_supremacy_circuit.py [hunk 1]
```diff
@@ -105,6 +105,41 @@ def google_v2_supremacy_circuit_grid(n_rows: int, n_cols: int,
...
+        n_rows: number of rows in a Bristlecone lattice.
+          Note that we do not include single qubit corners.
+        cz_depth: number of layers with CZ gates.
+        seed: seed for the random instance.
+
+    Returns:
+        A circuit with given size and seed.
+    """
+    def get_qubits(n_rows):
+        half_rows = n_rows // 2
```
Comment (Strilanc) [inline cirq/experiments/google_v2_supremacy_circuit.py]:
```python
min_row = 5 - n_rows // 2
max_row = min_row + n_rows - 1
dev = cirq.google.Bristlecone
qubits = [q for q in dev.qubits
          if min_row <= q.row <= max_row and len(dev.neighbors_of(q)) > 1]
```

File: cirq/experiments/google_v2_supremacy_circuit_test.py [hunk 2]
```diff
@@ -21,10 +21,27 @@ def test_google_v2_supremacy_circuit():
...
     assert circuit.operation_at(qubits[0], 1).gate == ops.CZ
     assert circuit.operation_at(qubits[5], 2).gate == ops.CZ
     assert circuit.operation_at(qubits[8], 3).gate == ops.CZ
     assert circuit.operation_at(qubits[13], 4).gate == ops.CZ
     assert circuit.operation_at(qubits[12], 5).gate == ops.CZ
     assert circuit.operation_at(qubits[13], 6).gate == ops.CZ
     assert circuit.operation_at(qubits[14], 7).gate == ops.CZ
+
+
+def test_google_v2_supremacy_bristlecone():
```
Comment (Strilanc) [inline cirq/experiments/google_v2_supremacy_circuit_test.py]: Do another test using fewer than the maximum number of rows.

File: cirq/experiments/google_v2_supremacy_circuit.py [hunk 3]
```diff
@@ -105,6 +105,41 @@ def google_v2_supremacy_circuit_grid(n_rows: int, n_cols: int,
...
+    Args:
+        n_rows: number of rows in a Bristlecone lattice.
+          Note that we do not include single qubit corners.
+        cz_depth: number of layers with CZ gates.
+        seed: seed for the random instance.
+
+    Returns:
+        A circuit with given size and seed.
+    """
+    def get_qubits(n_rows):
```
Comment (Strilanc) [inline cirq/experiments/google_v2_supremacy_circuit.py]: Check that `1 <= n_rows <= 11`.

File: cirq/experiments/google_v2_supremacy_circuit.py (hunk 1 above)
Comment (sboixo) [reply cirq/experiments/google_v2_supremacy_circuit.py]: sub-brislecone

The qubits I want for n_rows = 9 are inside the blue shape, and for n_rows = 10 are in the inside the red shape.

Comment (Strilanc) [reply cirq/experiments/google_v2_supremacy_circuit.py]: Oh, I've misunderstood the rule. Could you label the other sizes as well? You might want to include such diagrams in the code as docs. Is it always nearly a square?

File: cirq/experiments/google_v2_supremacy_circuit_test.py (hunk 2 above)
Comment (sboixo) [reply cirq/experiments/google_v2_supremacy_circuit_test.py]: Done

File: cirq/experiments/google_v2_supremacy_circuit.py (hunk 3 above)
Comment (sboixo) [reply cirq/experiments/google_v2_supremacy_circuit.py]: Done
